lib/searchable_by/concern.rb

```diff
@@ -32,14 +32,13 @@ def search_by(query)
       columns.each do |col|
         col.node ||= col.attr.is_a?(Proc) ? col.attr.call : arel_table[col.attr]
       end
-      clauses = Util.build_clauses(columns, values)
+
+      clauses = Util.build_full_clauses(columns.select(&:full_match?), query)
+      clauses << Util.build_partial_clauses(columns.select(&:partial_match?), values)&.inject(&:and)
```

Comment:
that's not a good way to do this, the problem is that you are completely missing all columns that are neither full nor partial match (which can theoretically happen if you look at the conditions, it's not an exclusive either/or)